lib/plugins/chuck-norris/index.js

```diff
@@ -0,0 +1,27 @@
+const Discord = require('discord.js')
+const { PLUGIN_NAME } = require('../symbols')
+
+const fetch = require('node-fetch')
+const embed = new Discord.MessageEmbed()
+const url = 'https://api.chucknorris.io/jokes/random'
+
+async function pluginChuckNorrisJokes (client) {
+  client.on('message', msg => {
+    if (msg.content === '!chucknorris') {
+      fetch(url)
+        .then(response => response.json())
+        .then(data => {
+          embed
+            .setTitle('Best of Chuck Norris jokes:')
+            .setColor(0xf5f5f5)
+            .setDescription(`${data.value} :laughing:`)
+          msg.channel.send(embed)
+        })
+        .catch(err => console.log(err))
+    }
+  })
```

Comment:
Itt több dolgot is megvariálnék. Tisztább ha rögtön visszatérünk abban az esetben ha nem a "parancsot" kaptuk meg. Így nem kell az egész kódot bentébb vinni. Promise helyett meg inkább async-await.

Az embed példányosítást behúztam ide, őszintén nem tudom mi zajlik magában a Discord.js-ben szóval az a biztos ha mindig új objektumból dolgozunk. Amúgy sem nyerünk szinte semmit ha ugyanazt a példányt használjuk.

```suggestion
    if (msg.content !== '!chucknorris') {
      return
    }

    try {
      const response = await fetch(API_URL)
      const data = await response.json()

      const embed = new Discord.MessageEmbed()
        .setTitle('Best of Chuck Norris jokes:')
        .setColor(0xf5f5f5)
        .setDescription(`${data.value} :laughing:`)

      msg.channel.send(embed)
    } catch (err) {
      console.error(err)
    }
```

Reply:
- embed-en én is agyaltam, azért lett aztán külön mert működött a tesztkörnyezetben, de igen biztosabb így
- az első változat async-await volt, úgy látszik az első elgondolás lett volna a jó :)